Functions/Mock.ps1

```diff
@@ -338,29 +338,7 @@ about_Mocking
 
         $mockTable["$ModuleName||$CommandName"] = $mock
 
-        $scriptBlock = { $ExecutionContext.InvokeProvider.Item.Set("Function:\script:$($args[0])", $args[1], $true, $true) }
```

Comment:
I extracted all this block the `Invoke-CommonSetupInMockScope` and then added the new functionality there in order to group common code and reduce the size of the `Mock` function`.